src/types/permissions.ts

```diff
@@ -42,33 +50,151 @@ export enum PermissionType {
 /**
  * Permission interfaces for different asset types
  */
-export interface FungibleAssetPermission {
-  type: PermissionType.FungibleAsset;
-  asset: string;
-  // Question: best type here?: number | string | bigint
-  balance: string;
+export class FungibleAssetPermission extends Serializable {
+  readonly type = PermissionType.FungibleAsset;
+
+  readonly asset: AccountAddress;
+
+  readonly balance: string;
+
+  constructor(asset: AccountAddress, balance: string) {
+    super();
+    this.asset = asset;
+    this.balance = balance;
+  }
+
+  static from({ asset, balance }: { asset: AccountAddress; balance: string }): FungibleAssetPermission {
+    return new FungibleAssetPermission(asset, balance);
+  }
```

Comment:
I prefer the single object as a parameter version, which do you all prefer?

Reply:
Yeah agreed, but it's a bit annoying with classes cause you have to write more code